### PR TITLE
fix: CIPipeline loading

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description
Migrating all the existing async requests in CIPipeline to async-await instead and having a loader for all calls in initial mount

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/CIPipelineN/CIPipeline.tsx [hunk 1]
```diff
@@ -191,67 +191,95 @@ export default function CIPipeline({
...
+                    list.push({
+                        id: env.id,
+                        clusterName: env.cluster_name,
+                        clusterId: env.cluster_id,
+                        name: env.environment_name,
+                        active: false,
+                        isClusterActive: env.isClusterActive,
+                        description: env.description,
+                    })
+                    const _selectedEnv = list.find((env) => env.id == envId)
```
Comment (vivek-devtron) [inline src/components/CIPipelineN/CIPipeline.tsx:273]: Either this find should be outside or there should be a check for envId direct. The setter must be outside of foreach

File: src/components/CIPipelineN/CIPipeline.tsx [hunk 2]
```diff
@@ -280,140 +308,152 @@ export default function CIPipeline({
...
 
-    // TODO: Move to async
-    const getInitialData = (): void => {
-        setPageState(ViewType.LOADING)
-        getSecurityModuleStatus()
-        if (ciPipelineId) {
-            // TODO: Would need to add loader for getAvailablePlugins to avoid state toggle
-            getInitDataWithCIPipeline(appId, ciPipelineId, true)
-                .then((ciResponse) => {
+    const validateStage = (stageName: string, _formData: PipelineFormType, formDataErrorObject?): void => {
```
Comment (vivek-devtron) [inline src/components/CIPipelineN/CIPipeline.tsx:311]: move these methods after the API call methods

Comment (AbhishekA1509) [reply src/components/CIPipelineN/CIPipeline.tsx:336]: Done

File: src/components/CIPipelineN/CIPipeline.tsx (hunk 1 above)
Comment (AbhishekA1509) [reply src/components/CIPipelineN/CIPipeline.tsx]: Done

**Pull request merged**